### PR TITLE
Verify gradle wrapper on gradle audit

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/audit/java/gradle.go [hunk 1]
```diff
@@ -40,10 +48,45 @@ func runGradle(buildConfiguration *utils.BuildConfiguration, excludeTestDeps, us
...
+	}
 	// Read config
 	vConfig, err := utils.ReadGradleConfig(configFilePath, useWrapper)
 	if err != nil {
 		return err
 	}
 	return gradleutils.RunGradle(vConfig, tasks, "", buildConfiguration, 0, useWrapper, true)
 }
+
+func verifyGradleWrapper() (bool, error) {
```
Comment (yahavi) [inline xray/audit/java/gradle.go:68]: IMO "verify" returns an error if the wrapper doesn't exist.
```suggestion
func isGradleWrapperExist() (bool, error) {
```

Comment (yahavi) [inline xray/audit/java/gradle.go:68]: Let's add a unit test for this function

File: xray/audit/java/gradle.go [hunk 2]
```diff
@@ -40,10 +48,45 @@ func runGradle(buildConfiguration *utils.BuildConfiguration, excludeTestDeps, us
...
+	if err != nil {
+		return false, err
+	}
+	for _, file := range filesInDir {
+		if file == fullPathGradlew || file == fullPathGradlewbat {
+			return true, nil
+		}
+	}
+
+	return false, nil
```
Comment (yahavi) [inline xray/audit/java/gradle.go:91]: Please consider the following:
```suggestion
	wrapperName := gradlew
	if coreutils.IsWindows() {
		wrapperName += ".bat"
	}
	return fileutils.IsFileExists(wrapperName, false)
}
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Comment by yahavi** [comment]
Also - there is an edge case we may consider - 
I'm not sure it is applicable in this scenario, but it is possible to provide `-p` or `--project-dir` flag to Gradle, to allow running build from another directory.

File: xray/audit/java/gradle.go (hunk 1 above)
Comment (omerzi) [reply xray/audit/java/gradle.go]: done

File: xray/audit/java/gradle.go (hunk 2 above)
Comment (omerzi) [reply xray/audit/java/gradle.go]: done

File: xray/audit/java/gradle.go (hunk 1 above)
Comment (omerzi) [reply xray/audit/java/gradle.go]: done

File: xray/audit/java/gradle.go [hunk 3]
```diff
@@ -2,13 +2,18 @@ package java
...
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
 	gradleutils "github.com/jfrog/jfrog-cli-core/v2/utils/gradle"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
 	"github.com/jfrog/jfrog-client-go/xray/services"
 )
 
+const (
+	gradlew = "gradlew"
+)
```
Comment (yahavi) [inline xray/audit/java/gradle.go:15]:
```suggestion
const gradlew = "gradlew"
```

Comment (omerzi) [reply xray/audit/java/gradle.go]: done

File: xray/audit/java/gradle.go [hunk 4]
```diff
@@ -40,10 +43,29 @@ func runGradle(buildConfiguration *utils.BuildConfiguration, excludeTestDeps, us
 			log.Debug("Using resolver config from", configFilePath)
 		}
 	}
+	// check if gradle wrapper exist
```
Comment (eyalbe4) [inline xray/audit/java/gradle.go:46]: check --> Check
if --> whether
exist --> exists

File: xray/audit/java/gradle.go [hunk 5]
```diff
@@ -40,10 +43,29 @@ func runGradle(buildConfiguration *utils.BuildConfiguration, excludeTestDeps, us
...
+	if useWrapper {
+		var wrapperExist bool
+		wrapperExist, err = isGradleWrapperExist()
+		if err != nil {
+			return
+		}
+		useWrapper = wrapperExist
+	}
```
Comment (eyalbe4) [inline xray/audit/java/gradle.go:54]: The above code can be simplified to -
```
	if useWrapper {
		useWrapper, err = isGradleWrapperExist()
		if err != nil {
			return
		}
	}
```

File: xray/audit/java/gradle.go [hunk 6]
```diff
@@ -40,10 +43,29 @@ func runGradle(buildConfiguration *utils.BuildConfiguration, excludeTestDeps, us
...
 	// Read config
 	vConfig, err := utils.ReadGradleConfig(configFilePath, useWrapper)
 	if err != nil {
 		return err
 	}
 	return gradleutils.RunGradle(vConfig, tasks, "", buildConfiguration, 0, useWrapper, true)
 }
+
+// This function assumes that the Gradle wrapper is in the root directory.
+// Adapting this function is needed if the audit command supports Gradle's --project-dir option.
```
Comment (eyalbe4) [inline xray/audit/java/gradle.go:64]: "Adapting this function is needed" - not sure what this means. Which adaptation are you referring to?

File: xray/audit/java/gradle.go (hunk 4 above)
Comment (omerzi) [reply xray/audit/java/gradle.go:46]: done

File: xray/audit/java/gradle.go (hunk 5 above)
Comment (omerzi) [reply xray/audit/java/gradle.go:54]: done

**Pull request merged**